keycloak/keycloak-admin-ui
# Run as theme

## Motivation
Allow new admin console to run as a theme on Keycloak server.

## Brief Description
When you build the app, it will no longer go into `/build`.  Instead, it will be built into `/build/src/main/resources/admin/resources`.  The reason for this change is that we now have a Maven `pom.xml` whose home is `/build`.  This Maven build will facilitate integration in the main Keycloak server build.

## Verification Steps
_Note: The steps below are for verification of Keycloak server integration.  Everything should work normally in development mode._
1.  `npm run build`
2. `cd build`
3. `mvn clean install`
4. copy `/build/target/classes/admin` directory to `<keycloak install>/themes/keycloak.v2`.  Alternatively, you can unzip keycloak-`admin-ui-0.0.1-SNAPSHOT.jar` into that directory.
5. Until we make changes to the Keycloak server, you will need to edit `<keycloak install>/themes/keycloak.v2/admin/index.ftl`.  You will need to hard-code the value for loginRealm.  So most of the time, it will be sufficient to replace `"${loginRealm}"` with `"master"`.
6. Go to `Realm Settings --> Themes` and set Admin Console Theme to `keycloak.v2`.
7. Go directly to Keycloak server with a URL such as [http://localhost:8180/auth/admin/master/console/](http://localhost:8180/auth/admin/master/console/)
8. Step 7 gets you to the console logging into master realm.  But you can also act as an administrator who is restricted to a single realm.  For that, you will need to set the name of the single realm, such as `"test"` in step 5.  Then create a user with proper permissions for the realm and log in with a URL such as [http://localhost:8180/auth/admin/test/console/](http://localhost:8180/auth/admin/test/console/).  See [https://www.keycloak.org/docs/6.0/server_admin/#_per_realm_admin_permissions](https://www.keycloak.org/docs/6.0/server_admin/#_per_realm_admin_permissions)

_You can also test step 8 in development mode.  Follow the same instructions, except the URL will be [http://localhost:8080/?realm=test](http://localhost:8080/?realm=test)_


## Checklist:

- [NA] User-visible strings are using the react-i18next framework (useTranslation)
- [NA] Help has been implemented
- [NA] axe report has been run and resulting a11y issues have been resolved
- [X] Type checking has been performed via 'npm run check-types'

## Additional Notes
One of the notes in the README.md does not apply yet.  `-Dprofile.feature.newadmin=enabled` does not exist in the Keycloak server yet.  So you do not need to enable it.
